src/vmm/src/cpu_config/templates.rs

```diff
@@ -195,26 +195,62 @@ where
     where
         D: Deserializer<'de>,
     {
-        let mut bitmap_str = String::deserialize(deserializer)?;
+        let original_str = String::deserialize(deserializer)?;
```

Comment:
This appears to be simply de-serializing 1 string into 2 integers, in this case I think this can be greatly simplified into something like: [playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=074c919a92fe2ed15682df87832baf68)
```rust
#[derive(Debug)]
struct UnknownCharacterInBitField;
impl std::fmt::Display for UnknownCharacterInBitField {
    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
        write!(f, "Unknown character found in bitfield.")
    }
}
impl std::error::Error for UnknownCharacterInBitField {}

fn main() -> Result<(),Box<dyn std::error::Error>> {
    let original_str = String::from("0bx00100x0x1xxxx01xxx1xxxxxxxxxxx1");

    let stripped_str = original_str.strip_prefix("0b").unwrap_or(&original_str);

    let (mut filter,mut value) = (0u64,0u64);
    let mut i = 0;
    for s in stripped_str.as_bytes().iter().rev() {
        match s {
            b'_' => continue,
            b'x' => (),
            b'0' => {
                filter |= 1 << i;
            },
            b'1' => {
                value |= 1 << i;
            },
            _ => return Err(Box::new(UnknownCharacterInBitField)),
        }
        i += 1;
    }
    println!("filter: {filter:x}");
    println!("value:  {value:x}");
    Ok(())
}
```

Reply:
Version with improved generics https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c0a7e03edb76d40e62632678785ba1e6